applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationClientTest.java

```diff
@@ -579,19 +846,70 @@ public void deleteAllSettings() {
                     }
                 }).blockLast();
     }
+    
+    /**
+     * Test the API will not make a get call without having a key passed, an IllegalArgumentException should be thrown.
+     */
+    @Test
+    public void getSettingRequiresKey() {
+        assertRunnableThrowsArgumentException(() -> client.get(null));
+        assertRunnableThrowsArgumentException(() -> client.get(""));
+    }
+    
+    /**
+     * Test the API will not make a delete call without having a key passed, an IllegalArgumentException should be thrown.
+     */
+    @Test
+    public void deleteSettingRequiresKey() {
+        assertRunnableThrowsArgumentException(() -> client.delete(null));
+        assertRunnableThrowsArgumentException(() -> client.delete(null, null, null));
+        assertRunnableThrowsArgumentException(() -> client.delete(""));
+        assertRunnableThrowsArgumentException(() -> client.delete("", null, null));
+    }
+    
+    /**
+     * Test the API will not make lock or unlock calls without having a key passed, an IllegalArgumentException should be thrown.
+     */
+    @Test
+    public void lockAndUnlockRequiresKey() {
+        assertRunnableThrowsArgumentException(() -> client.lock(null));
```

Comment:
this is where a @theory class would be nice because we don't need the session-records. Would it be possible to create a separate class that just runs these?